aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -311,7 +309,7 @@ impl Execution {
             v2_flag,
         );
         if self.execution_mode.is_compare() {
-            let res_other_opt = self.execute_code(
+            let res_other = self.execute_code(
                 cur_version,
                 state,
```

Comment:
how does this work now if the passed state is not mut but execute_code expects mut?

Reply:
it doesn't expect mutable, it takes ownership of the object. 

when you take ownership, you can decide if you want mut or not, because no one else has access to it

Reply:
ah right